src/DIRAC/WorkloadManagementSystem/Utilities/Runner.py

```diff
@@ -0,0 +1,137 @@
+import os
+import shlex
+import time
+
+from DIRAC import gLogger, gConfig, S_OK, S_ERROR
+from DIRAC.Core.Utilities.Subprocess import systemCall
+from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+from DIRAC.Core.Utilities.LockRing import LockRing
+from DIRAC.Resources.Computing.ComputingElementFactory import ComputingElementFactory
+from DIRAC.ConfigurationSystem.Client.Helpers.Resources import getQueue
+
+
+class Runner(object):
+
+  def __init__(self):
+    self.log = gLogger.getSubLogger('Runner')
+
+  def run(self, command, environment=None, callbackFunction=None, bufferLimit=52428800):
+    workloadExecutionLocation = os.environ.get('WORKLOADEXECLOCATION')
+
+    # if workloadExecutionLocation is not empty, this means the workload should be executed
+    # in a different remote location. This mainly happens when the remote Site has no
+    # external connectivity and can only execute the workload itself.
+    if workloadExecutionLocation:
+      return self._remoteRun(remoteSite=workloadExecutionLocation,
+                             command=command)
+
+    return self._localRun(command=command,
+                          environment=environment,
+                          callbackFunction=callbackFunction,
+                          bufferLimit=bufferLimit)
+
+  def _localRun(self, command, environment=None, callbackFunction=None, bufferLimit=0):
```

Comment:
There is no need for this method.

Reply:
Yes indeed, I added it to have only one line to add/remove in the modules that execute applications/scripts such as:

```python
return Runner().run(command=command, environment=env, callbackFunction=self.__redirectLogOutput)
```

Would you prefer something like this (I would rename `Runner` in this case of course)?

```python
result = Runner().is_remote_execution()
if result:
    result = Runner().remoteRun(command)
else:
    result = systemCall(command, ...)
    ...
```

Reply:
yes, the latter code would be simpler.